src/js/samples/components/Detail/SampleFileSizeWarning.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { Alert, Icon } from "../../../base";
+
+type SampleFileSizeWarningProps = {
+    sampleId: string;
+    show: boolean;
+    showLink: boolean;
+};
+
+/**
+ * Displays a warning if any sample files are under 10 megabytes
+ */
+export default function SampleFileSizeWarning({ sampleId, show, showLink }: SampleFileSizeWarningProps) {
```

Comment:
I would be tempted to rethink the interface here. I think it is within the scope of `SampleFileSizeWarning` to be passed the file size and determine internally if it is large enough to render the warning. Also you can probably also handle showLink internally as well given that it just uses `location`